falcon/request.py

```diff
@@ -1028,6 +1059,22 @@ def _parse_form_urlencoded(self):
 
             self._params.update(extra_params)
 
+    def _parse_rfc_forwarded(self):
+        """rfc 7239"""
+        addr = []
+        for forwarded in self.env['HTTP_FORWARDED'].split(','):
+            for param in forwarded.split(';'):
+                param = param.strip().split('=', 1)
+                if len(param) == 1:
+                    continue
+                key, val = param
+                if key.lower() not in ('for', 'by'):
+                    # we only want for/by params
+                    continue
+                host, _ = parse_host(unquote_string(val))
```

Comment:
Well after I did an SNS optimization to unquote_string it becomes faster than werkzeug (in most time):

```python
from timeit import timeit
timeit('unquote(r\'"127.0.0.1"\')', 'from falcon.util.uri import unquote_string as unquote')
# 0.24731779098510742
timeit('unquote(r\'"127.0.0.1"\')', 'from werkzeug.http import unquote_header_value as unquote')
# 0.42057013511657715
timeit('unquote(r\'"127.0.0.1"\')', 'from falcon.util.uri import unquote_string_old as unquote')
# 0.6000280380249023
```

And even quoted `"`:

```python
timeit('unquote(r\'"\\"unknown\\""\')', 'from falcon.util.uri import unquote_string as unquote')
# 0.4324629306793213
timeit('unquote(r\'"\\"unknown\\""\')', 'from werkzeug.http import unquote_header_value as unquote')
# 0.5061039924621582
timeit('unquote(r\'"\\"unknown\\""\')', 'from falcon.util.uri import unquote_string_old as unquote')
# 0.6474518775939941
```

Well only the double backslash `\` are slower:

```python
timeit('unquote(r\'"bad\\\\"\')', 'from falcon.util.uri import unquote_string as unquote')
# 0.8454480171203613
timeit('unquote(r\'"bad\\\\"\')', 'from werkzeug.http import unquote_header_value as unquote')
# 0.47409915924072266
timeit('unquote(r\'"bad\\\\"\')', 'from falcon.util.uri import unquote_string_old as unquote')
# 0.7625319957733154
```